debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/ReplicaIdentityMapper.java

```diff
@@ -0,0 +1,110 @@
+            return this.replicaIdentity;
+        }
+
+        public String getIndexName() {
+            return this.indexName;
+        }
+
+        @Override
+        public boolean equals(Object replicaIdentityOption) {
+            if (replicaIdentityOption instanceof ReplicaIdentityValue) {
+                ReplicaIdentityValue replicaIdentityValueObj = (ReplicaIdentityValue) replicaIdentityOption;
+                return replicaIdentityValueObj.getReplicaIdentity().equals(this.getReplicaIdentity())
+                        && (replicaIdentityValueObj.getIndexName() != null && replicaIdentityValueObj.getIndexName().equals(this.indexName)
+                                || (!this.replicaIdentity.equals(ServerInfo.ReplicaIdentity.INDEX)));
+            }
+            return false;
+        }
+
+        @Override
+        public String toString() {
+            return this.replicaIdentity == ServerInfo.ReplicaIdentity.INDEX ? String.format("%s %s", this.replicaIdentity, this.indexName)
+                    : this.replicaIdentity.toString();
+        }
+
+        @Override
+        public int hashCode() {
+            int result = 17;
+            if (this.replicaIdentity != null) {
+                result = 31 * result + this.replicaIdentity.hashCode();
+            }
+            if (this.indexName != null) {
+                result = 31 * result + this.indexName.hashCode();
+            }
+            return result;
+        }
+
+    }
+
+    public static final Pattern REPLICA_AUTO_SET_PATTERN = Pattern
+            .compile("(?i)^\\s*(?<tablename>[^\\s:]+):(?<replicaidentity>DEFAULT|(INDEX) (?<indexname>.\\w*)|FULL|NOTHING)\\s*$");
+    public static final Pattern PATTERN_SPLIT = Pattern.compile(";");
+
+    private final Map<TableId, ReplicaIdentityValue> replicaIdentityMapper;
+
+    public ReplicaIdentityMapper(String databaseName, String replicaAutoSetValue) {
+        this.replicaIdentityMapper = getReplicaIdentityMapper(databaseName, replicaAutoSetValue);
+    }
+
+    private Map<TableId, ReplicaIdentityValue> getReplicaIdentityMapper(String databaseName, String replicaAutoSetValue) {
+
+        if (replicaAutoSetValue == null) {
+            return null;
+        }
+
+        return Arrays.stream(PATTERN_SPLIT.split(replicaAutoSetValue))
+                .map(REPLICA_AUTO_SET_PATTERN::matcher)
+                .filter(Matcher::matches)
+                .collect(Collectors.toMap(
+                        t -> {
+                            String[] tableName = t.group("tablename").split("\\.");
+                            return new TableId(databaseName, tableName[0], tableName[1]);
+                        },
+                        t -> new ReplicaIdentityValue(
+                                ServerInfo.ReplicaIdentity.valueOf(t.group("replicaidentity").replaceAll("\\s.*", "")), t.group("indexname"))));
+    }
+
+    public Map<TableId, ReplicaIdentityValue> getMapper() {
```

Comment:
I think the naming really needs to be improved. The whole class that manages mapping is named mapper and this return map which is an internal data. Cannot this class provide the business logic needed so it is not necessary to expaose the map?

Reply:
Thanks @jpechane we think this makes sense. So just to check what you are proposing is: 
- Convert `io.debezium.connector.postgresql.connection.ServerInfo.ReplicaIdentity` to a standard class which includes an enum for the different replica identity options (`FULL`, `DEFAULT`, etc) and a string representing the index field for the replica identity when the option is `INDEX`
- Migrate to this class the current business logic in `ReplicaIdentityMapper` which will remove the need to expose the `Map<TableId, ReplicaIdentityValue>` map.

Let us know if that's right in the right direction and we'll update the PR accordingly.

cc @miguelbirdie

Reply:
Yes, I beleive we are on the same page.